BootloaderCommonPkg/Include/Library/IasImageLib.h

```diff
@@ -144,7 +144,7 @@ IasGetFiles (
 #define IAS_PAYLOAD_CRC(h)            (* (UINT32*) (((UINT32*) (h)) + (h)->DataOffset + (h)->DataLength))
 #define IAS_PAYLOAD_END(h)            (IAS_PAYLOAD(h) + (h)->DataLength + sizeof(UINT32))
 #define IAS_SIGNATURE(h)              (((UINT32)(UINTN)h) + ROUNDED_UP((h)->DataOffset + (h)->DataLength + sizeof(UINT32), 256))
-#define IAS_PUBLIC_KEY(h)             ((IAS_SIGNATURE(h) + 256))
+#define IAS_PUBLIC_KEY(h)             (((( (h)->ImageType  & 1<<8 ) == (1 <<8) ? IAS_SIGNATURE(h) + 256 : IAS_PAYLOAD_END(h))))

```

Comment:
Can we use macro BIT8 for (1<<8) ?   It is already defined in base include files.

Reply:
Applied :)

Reply:
Thank you!